### ISSUE TITLE
Introduce a Success Criteria object to reporting

### ISSUE THREAD

**Issue by jessebeach** [quailjs/quail#140]
Given an entry in the WCAG guideline like this, we'd like to extend it to include the combination of techniques (and their logical relationships) that would constitute Sufficient Techniques to declare the Criteria satisfied.

``` yaml
2.4.1: 
  id: "navigation-mechanisms-skip"
  title: "Bypass Blocks"
  description: "A mechanism is available to bypass blocks of content that are repeated on multiple Web pages."
  uri: "http://www.w3.org/TR/WCAG20/#navigation-mechanisms-skip"
  techniques: 
    - "C6"
    - "G1"
    - "G123"
    - "G124"
    - "H64"
    - "H69"
    - "H70"
    - "SCR28"
    - "SL25"
    - "SL29"
```

We would extend it to something like

``` yaml
2.4.1: 
  id: "navigation-mechanisms-skip"
  title: "Bypass Blocks"
  description: "A mechanism is available to bypass blocks of content that are repeated on multiple Web pages."
  uri: "http://www.w3.org/TR/WCAG20/#navigation-mechanisms-skip"
  techniques: 
    - "C6"
    - "G1"
    - "G123"
    - "G124"
    - "H64"
    - "H69"
    - "H70"
    - "SCR28"
    - "SL25"
    - "SL29"
  success:
    - "G1 OR G123 OR G124 OR SL125 OR H69 OR H70 OR SCR28"
```

Success should be determined as the logical relationship between the state of Techniques for a given context. The relationships can be related through OR and AND operators. 

We'll need to deal with the concept of "testability" here, which allows a Technique to land in between Pass and Fail.

This issue is related to: #101, #99, #97
